// [[Reason: Cannot resolve non-static property names]] ////

//// [tests/cases/compiler/declarationEmitLateBoundAssignments2.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -5,8 +5,9 @@
 export declare namespace decl {
     var B: string;
 }
 export declare function decl2(): void;
+export declare namespace decl2 { }
 export declare namespace decl2 {
     var C: number;
 }
 export declare function decl3(): void;
@@ -69,11 +70,20 @@
 /// [Errors] ////
 
 declarationEmitLateBoundAssignments2.ts(17,9): error TS2733: Property 'C' was also declared here.
 declarationEmitLateBoundAssignments2.ts(19,1): error TS2718: Duplicate property 'C'.
+declarationEmitLateBoundAssignments2.ts(19,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitLateBoundAssignments2.ts(22,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitLateBoundAssignments2.ts(25,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitLateBoundAssignments2.ts(28,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitLateBoundAssignments2.ts(31,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitLateBoundAssignments2.ts(34,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitLateBoundAssignments2.ts(37,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitLateBoundAssignments2.ts(40,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitLateBoundAssignments2.ts(43,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
 
 
-==== declarationEmitLateBoundAssignments2.ts (2 errors) ====
+==== declarationEmitLateBoundAssignments2.ts (11 errors) ====
     // https://github.com/microsoft/TypeScript/issues/54811
     
     const c = "C"
     const num = 1
@@ -95,32 +105,50 @@
     }
     decl2[c] = 0
     ~~~~~~~~
 !!! error TS2718: Duplicate property 'C'.
+    ~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export function decl3(): void {}
     decl3[77] = 0
+    ~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export function decl4(): void {}
     decl4[num] = 0
+    ~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export function decl5(): void {}
     decl5["101"] = 0
+    ~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export function decl6(): void {}
     decl6[numStr] = 0
+    ~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export function decl7(): void {}
     decl7["qwe rty"] = 0
+    ~~~~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export function decl8(): void {}
     decl8[withWhitespace] = 0
+    ~~~~~~~~~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export function decl9(): void {}
     decl9["ðŸ¤ª"] = 0
+    ~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export function decl10(): void {}
     decl10[emoji] = 0
+    ~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     export const arrow: {
         (): void
         B: string
